vm/src/obj/objtuple.rs

```diff
@@ -238,11 +238,15 @@ impl PyTuple {
         vm: &VirtualMachine,
     ) -> PyResult<PyTupleRef> {
         let elements = if let OptionalArg::Present(iterable) = iterable {
-            if iterable.lease_class().is(&cls) {
-                return Ok(PyRef::from_obj_unchecked(iterable));
+            let exact_res = if cls.is(&vm.ctx.types.tuple_type) {
+                iterable.downcast_exact::<PyTuple>(vm)
+            } else {
+                Err(iterable)
+            };
+            match exact_res {
+                Ok(tup) => return Ok(tup),
+                Err(iterable) => vm.extract_elements(&iterable)?,
```

Comment:
This code seems confusing at the first glance maybe because the type of exact_res is not PyResult.
How about avoiding to bind non-PyResult to variable?
```suggestion
            let iterable = if cls.is(&vm.ctx.types.tuple_type) {
                match iterable.downcast_exact::<PyTuple>(vm) {
                    Ok(tuple) => return Ok(tuple),
                    Err(iterable) => iterable,
                }
            } else {
                iterable
            };
            vm.extract_elements(&iterable)?
```